### PR TITLE
feat(react-components): MeasurementButton with URL state

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:

Another stepping stone to enable storage of measurements in the URL in FDX.

Depends on https://github.com/cognitedata/reveal/pull/3936

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/components/RevealToolbar/MeasurementButton.tsx [hunk 1]
```diff
@@ -74,3 +86,8 @@ export const MeasurementButton = (): ReactElement => {
     </CogsTooltip>
   );
 };
+
+const useUpdateMeasurementsCallback = (
```
Comment (pramodcog) [inline react-components/src/components/RevealToolbar/MeasurementButton.tsx:90]: Unused code

File: react-components/src/components/RevealToolbar/MeasurementButton.tsx [hunk 2]
```diff
@@ -2,43 +2,55 @@
  * Copyright 2023 Cognite AS
  */
 
-import { useMemo, type ReactElement, useState, useEffect } from 'react';
+import { type ReactElement, useState, useEffect, useCallback } from 'react';
 import { useReveal } from '../RevealContainer/RevealContext';
 import { Button, Tooltip as CogsTooltip } from '@cognite/cogs.js';
-import { MeasurementTool } from '@cognite/reveal/tools';
+import { type Measurement, MeasurementTool } from '@cognite/reveal/tools';
```
Comment (pramodcog) [inline react-components/src/components/RevealToolbar/MeasurementButton.tsx:8]:
```suggestion
import { type Measurement, type MeasurementTool } from '@cognite/reveal/tools';
```

File: react-components/src/components/RevealToolbar/measurementUrlStateHooks.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/*!
```
Comment (pramodcog) [inline react-components/src/components/RevealToolbar/measurementUrlStateHooks.ts:1]: Can we move file to `hooks` folder

**Review by pramodcog — CHANGES_REQUESTED** [review]
Works well, minor comments

File: react-components/src/components/RevealToolbar/measurementUrlStateHooks.ts (hunk 3 above)
Comment (haakonflatval-cognite) [reply react-components/src/components/RevealToolbar/measurementUrlStateHooks.ts]: Created a new `hooks` folder in the toolbar directory

**Review by pramodcog — APPROVED** [review]
Looks good

**Pull request merged**